asyncgit/src/sync/config.rs

```diff
@@ -62,6 +62,54 @@ pub fn untracked_files_config_repo(
 	Ok(ShowUntrackedFilesConfig::All)
 }
 
+// see https://git-scm.com/docs/git-config#Documentation/git-config.txt-pushdefault
+/// represent `push.default` git config
+pub enum PushDefaultStrategyConfig {
+	Nothing,
+	Current,
+	Upstream,
+	Simple,
+	Matching,
+}
+
+impl Default for PushDefaultStrategyConfig {
+	fn default() -> Self {
+		Self::Simple
+	}
+}
+
+impl<'a> TryFrom<&'a str> for PushDefaultStrategyConfig {
+	type Error = crate::Error;
+	fn try_from(
+		value: &'a str,
+	) -> std::result::Result<Self, Self::Error> {
+		match value {
+			"nothing" => Ok(Self::Nothing),
+			"current" => Ok(Self::Current),
+			"upstream" => Ok(Self::Upstream),
+			"simple" => Ok(Self::Simple),
+			"matching" => Ok(Self::Matching),
+			_ => Err(crate::Error::Generic(format!(
+				"{value} is incorrect push.default"
```

Comment:
Let's steal the error message from git. This makes it easier for users to find a solution.

```suggestion
				"malformed value for push.default: {value}, must be one of nothing, matching, simple, upstream or current"
```

Reply:
Good point, like it!